pkg/pool-hooks/contracts/utils/StableSurgeMedianMath.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+library StableSurgeMedianMath {
+    function calculateImbalance(uint256[] memory balances) internal pure returns (uint256) {
+        uint256[] memory sortedBalances = sort(balances);
+        uint256 median = findMedian(sortedBalances);
+
+        uint256 totalBalance = 0;
+        uint256 totalDiff = 0;
+
+        for (uint i = 0; i < balances.length; i++) {
+            totalBalance += balances[i];
+            totalDiff += absSub(balances[i], median);
+        }
+
+        return (totalDiff * 1e18) / totalBalance;
+    }
+
+    function findMedian(uint256[] memory sortedBalances) internal pure returns (uint256) {
+        uint256 mid = sortedBalances.length / 2;
+
+        if (sortedBalances.length % 2 == 0) {
+            return (sortedBalances[mid - 1] + sortedBalances[mid]) / 2;
+        } else {
+            return sortedBalances[mid];
+        }
+    }
+
+    // First implementation had quickselect algorithm, but it was removed in the final version
+    // because it seems to be overhead for small arrays (up to 8 elements)
+    // Complexity of this insertion sort is O(n^2) but it is not a problem for small arrays
+    function sort(uint256[] memory balances) internal pure returns (uint256[] memory sortedBalances) {
```

Comment:
Have you considered this? https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Arrays.sol#L43

Unless this implementation has a significant advantage, I'd say let's just use OZ for basic functionality.